h2o-automl/src/main/java/ai/h2o/automl/targetencoding/TargetEncoderModelAlgoRegister.java

```diff
@@ -0,0 +1,20 @@
+package ai.h2o.automl.targetencoding;
+
+import hex.ModelBuilder;
+import water.api.AlgoAbstractRegister;
+import water.api.RestApiContext;
+import water.api.SchemaServer;
+
+public class TargetEncoderModelAlgoRegister extends AlgoAbstractRegister {
+  
+  @Override
+  public void registerEndPoints(RestApiContext context) {
+    ModelBuilder targetEncoderModelBuilder = new TargetEncoderBuilder(true);
+    registerModelBuilder(context, targetEncoderModelBuilder, SchemaServer.getStableVersion());
```

Comment:
why not use RegisterAlgos?

Reply:
Not available there, would require playing with dependencies. But your comment raises another question:

We have this functionality in the AutoML module and the rest of AutoML behaves the same way (acts as an extension, even though our core modules directly depend on it and could be considered part of core) - it is registered as every H2O extension is.

However, even though Target Encoding was primarily meant for AutoML, it makes sense to me to relocate it to core algos (can be done any day), as it's definitely not tied to AutoML and usable **independently** by our users.

WDYT everoyne ?

Reply:
exactly my thinging, if we are exposing this as a regular algo it should live in core

Reply:
Made sense to me from the beginning. Let's make this bold move and make it part of H2O Core, as AutoML is of great importance. :)

Reply:
First, I need to make sure SW is ok with it @mn-mikke .

Reply:
Done. Sparkling Water is fine with it.